draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -448,12 +456,94 @@ Steps:
 20.  y = u * x (mod p)       // ux
 21.  y = y + v (mod p)       // ux + v
 22. Output (x, y)
-
 ~~~
 
 ### Shallue-Woestijne-Ulas Method {#swu}
 
-((TODO: write this section))
+The Shallue-Woestijne-Ulas (SWU) method, originated in part by
+Shallue and Woestijne {{SW06}} and later simplified and extended by Ulas {{SWU07}},
+deterministically encodes an artbirary string to a point on a curve. 
+This algorithm works for any curve over F_{p^n}. Given curve equation 
+g(x) = x^3 + Ax + B, two separate HashToBase implementations, H0 and H1, 
+this algorithm works as follows:
+
+~~~
+1. t = H0(alpha)
+2. u = H1(alpha)
+3. X1(t, u) = u
+4. X2(t, u) = (-B / A)(1 + 1 / (t^4 * g(u)^2 + t^2 * g(u)))
+5. X3(t, u) = t^3 * g(u)^2  * g(X2(t, u))
+6. If g(X1(t, u)) is square, output (X1(t, u), sqrt(g(X1(t, u))))
+7. If g(X2(t, u)) is square, output (X2(t, u), sqrt(g(X2(t, u))))
+8. Output (X3(t, u), sqrt(g(X3(t, u))))
+~~~
+
+The algorithm relies on the following equality:
+
+~~~
+t^3 * g(u)^2  * g(X2(t, u)) = g(X1(t, u)) * g(X2(t, u)) * g(X3(t, u))
+~~~
+
+Thus, at least one of X1(t, u), X2(t, u), X3(t, u) is the x-coordinate of a point
+on the curve. Computing the corresponding y value requires computing a square root.
```

Comment:
I think these details are a little too much. I think it is sufficient to say something like "The algorithm computes three candidate points, constructed such that at least one of them lies on the curve".